### ISSUE TITLE
Change `qr` bit

### ISSUE THREAD

**Issue by CrabNejonas** [NLnetLabs/domain#203]
The DNS packet header bit `QR` is set to `0` for queries and `1` for a response.
[`domain::base::header::Header`](https://docs.rs/domain/latest/domain/base/header/struct.Header.html#) casts this to a `bool` where `0` becomes `false` and `1` becomes `true`.

This is incredibly confusing to me since `qr` is clearly meant to mean `query`, so when you use the `qr()` method you basically have to to *the opposite* of what the header field indicates.

While I agree this is *technically* correct (as it just describes wether the flag is set) this took me about 1hr and largely re-implementing a dns parser from scratch to figure out. Soooo maybe we should do something about that 😄

**Comment by partim** [comment]
I agree that the definition of the QR bit is unfortunate. However, having a method that returns `false` when the bit is set only makes things worse. What I think we could do is adding a separate method `is_query` that does the flipping. The more elaborate name would likely cause people to pick that one over plain `qr`, so I think that is a good solution?

**Comment by CrabNejonas** [comment]
Yeah that could be a good idea, another idea I had was changing `qr()` to return an enum instead of a bool, so 
```rust
#[repr(u8)]
enum QR {
   Query = 0,
   Response = 1
}
```
that way it's impossible to get confused

**Comment by partim** [comment]
Hm, yes. But then this flag behaves differently from the other flags, which I kind of dislike. Also, `msg.is_query()` is shorter and perhaps more straightforward than `matches!(msg.qr(), Qr::Query)`.
